service/r2/service.go

```diff
@@ -128,29 +134,32 @@ func registerRoute(router *mux.Router, path, method string, h r2Handler, hf r2Ha
 
 // service handles all of the endpoints for r2.
 type service struct {
-	rootPrefix  string
-	store       Store
-	authService auth.HTTPAuthService
-	logger      log.Logger
-	nowFn       clock.NowFn
-	metrics     serviceMetrics
+	rootPrefix   string
+	store        store.Store
+	authService  auth.HTTPAuthService
+	logger       log.Logger
+	nowFn        clock.NowFn
+	metrics      serviceMetrics
+	updateHelper rules.RuleSetUpdateHelper
```

Comment:
Is this a helper or just update options?

Reply:
It is a helper. It appears to be this factory sturct that comes from m3metircs.